File: 2021/11/02/relative_sort_array.py
"""
https://leetcode.com/problems/relative-sort-array/
Author: Andrew Jarombek
Date: 11/2/2021
"""

from typing import List


def relative_sort_array(arr1: List[int], arr2: List[int]) -> List[int]:
    order = {value: index for index, value in enumerate(arr2)}
    return sorted(arr1, key=lambda x: order.get(x, 1001 + x))

File: 2022/09/30/relativeSortArray.ts
/**
 * https://leetcode.com/problems/relative-sort-array/
 * @author Andrew Jarombek
 * @since 9/30/2022
 */

function relativeSortArray(arr1: number[], arr2: number[]): number[] {
    let map = new Map<number, number>();

    for (let i = 0; i < arr2.length; i++) {
        let val = arr2[i];
        map.set(val, i);
    }

    arr1.sort((a, b) => {
        let val = (map.get(a) ?? arr1.length) - (map.get(b) ?? arr1.length);
        if (val == 0)
            val = a - b;

        return val;
    });
    return arr1;
}